sdk/clientcore/core/src/main/java/io/clientcore/core/credentials/oauth/OAuthTokenRequestOptions.java

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.credentials.oauth;
+
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * <p>
+ * The {@link OAuthTokenRequestOptions} is a class used to provide additional information and context when requesting an
+ * access token from an authentication source. It allows you to customize the token request and specify additional
+ * parameters, such as scopes, claims, or authentication options.
+ * </p>
+ *
+ * <p>
+ * Here's a high-level overview of how you can use the {@link OAuthTokenRequestOptions}:
+ * </p>
+ *
+ * <ol>
+ * <li>Create an instance of the {@link OAuthTokenRequestOptions} class and configure the required properties.
+ * The {@link OAuthTokenRequestOptions} class allows you to specify the scopes or resources for which you want to request
+ * an access token, as well as any additional properties.</li>
+ *
+ * <li>The authentication client or mechanism will handle the token request options and return an access token that can
+ * be used to authenticate.</li>
+ * </ol>
+ *
+ * @see io.clientcore.core.credentials
+ */
+public class OAuthTokenRequestOptions {
```

Comment:
I thought we decided to keep `TokenRequestContext` for conceptual alignment (as we are doing with `TokenCredential`)

Reply:
Yeah, was leaning towards options as this is broader than TRC, but to keep it conceptually consistent renamed it as OAuthTRC